sql/src/test/java/io/crate/integrationtests/OptimizeTableIntegrationTest.java

```diff
@@ -101,6 +101,24 @@ public void testOptimizeWithParams() throws Exception {
     }
 
     @Test
+    public void testOptimizeWithUpgradeSegments() throws Exception {
+        execute("create table test (id int primary key, name string)");
+        ensureYellow();
+        execute("insert into test (id, name) values (0, 'Trillian'), (1, 'Ford'), (2, 'Zaphod')");
+        execute("select count(*) from test");
+        assertThat((long) response.rows()[0][0], lessThanOrEqualTo(3L));
+
+        execute("delete from test where id=1");
+        refresh();
+        execute("optimize table test with (max_num_segments=1, only_expunge_deletes=true)");
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.rows().length, is(0));
+
+        execute("select count(*) from test");
+        assertThat(response.rows()[0][0], is(2L));
```

Comment:
this test seems not to test the `upgrade_segments` parameter as the test name implies to